### PR TITLE
Fixed bugs, added test cases and handling for some edge case scenarios.

### PR DESCRIPTION
**Pull request by saleem-latif**
_Description:__

This PR contains following changes

- Added tests for config file updates, updated logic so that async writes and locking wait do not cause the loss of data added in previous writes.
- Added more error handling for cloud watch logs so that it gracefully handles errors and does not interrupt users.
- Make sure thread waiting on lock release does not override the contents of previous write.

### CODE REVIEW

File: src/main/java/org/intellij/sdk/codesync/CodeSyncLogger.java [hunk 1]
```diff
@@ -144,6 +144,11 @@ private static void logEvent(String message, String userEmail, String type) {
                 accessKey = PLUGIN_USER_ACCESS_KEY;
                 secretKey = PLUGIN_USER_SECRET_KEY;
             }
+
+            if (accessKey == null || secretKey == null) {
+                System.out.println("Plugin is not able communicate with the server. Please check your internet connection.");
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/CodeSyncLogger.java:149]: Internet connection error?

**Review by Basit-qc — APPROVED** [review]

File: src/main/java/org/intellij/sdk/codesync/CodeSyncLogger.java (hunk 1 above)
Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/CodeSyncLogger.java:149]: Yes, If user is not connected to the internet then these will be `null`. I added this so that plugin does not error our and keeps tracking changes until user is back online and sync resumes.
